cli/src/action/api.rs

```diff
@@ -65,8 +65,15 @@ pub struct SplinterRestClient {
 impl SplinterRestClient {
     /// Gets the Splinter node's status.
     pub fn get_node_status(&self) -> Result<NodeStatus, CliError> {
-        Client::new()
-            .get(&format!("{}/status", self.url))
+        #[allow(unused_mut)]
```

Comment:
This should have a comment explaining why this lint is allowed; it should specifically mention the feature that, if removed, would allow us to remove this annotation. Same for the other occurrences in this PR.

Reply:
Added comments